pallets/multisig/src/lib.rs

```diff
@@ -735,75 +853,161 @@ impl<T: Trait> Module<T> {
+        current_did: IdentityId,
+    ) -> DispatchResult {
+        let approvals_needed = Self::ms_signs_required(multisig.clone());
+        if proposal_details.approvals >= approvals_needed {
+            let ms_key = AccountKey::try_from(multisig.clone().encode())?;
+            if let Some(did) = <Identity<T>>::get_identity(&ms_key) {
+                ensure!(<Identity<T>>::has_valid_cdd(did), Error::<T>::CddMissing);
+                T::CddHandler::set_current_identity(&did);
+            } else {
+                let creator_identity = Self::ms_creator(&multisig);
                 ensure!(
-                    T::ChargeTxFeeTarget::charge_fee(
-                        proposal.encode().len().try_into().unwrap_or_default(),
-                        proposal.get_dispatch_info(),
-                    )
-                    .is_ok(),
-                    Error::<T>::FailedToChargeFee
+                    <Identity<T>>::has_valid_cdd(creator_identity),
+                    Error::<T>::CddMissing
                 );
+                T::CddHandler::set_current_identity(&creator_identity);
+            }
+            ensure!(
+                T::ChargeTxFeeTarget::charge_fee(
+                    proposal.encode().len().try_into().unwrap_or_default(),
+                    proposal.get_dispatch_info(),
+                )
+                .is_ok(),
+                Error::<T>::FailedToChargeFee
+            );
 
-                <ProposalClosed<T>>::insert(multisig_proposal, true);
-                let res = match proposal
-                    .dispatch(frame_system::RawOrigin::Signed(multisig.clone()).into())
-                {
-                    Ok(_) => true,
+            let res =
+                match proposal.dispatch(frame_system::RawOrigin::Signed(multisig.clone()).into()) {
+                    Ok(_) => {
+                        proposal_details.status = ProposalStatus::ExecutionSuccessful;
+                        true
+                    }
                     Err(e) => {
                         let e: DispatchError = e;
                         sp_runtime::print(e);
+                        proposal_details.status = ProposalStatus::ExecutionFailed;
                         false
                     }
                 };
-                let current_did = Context::current_identity::<Identity<T>>()
-                    .ok_or_else(|| Error::<T>::MissingCurrentIdentity)?;
-                Self::deposit_event(RawEvent::ProposalExecuted(
-                    current_did,
-                    multisig,
-                    proposal_id,
-                    res,
-                ));
-                Ok(())
-            } else {
-                Ok(())
+            Self::deposit_event(RawEvent::ProposalExecuted(
+                current_did,
+                multisig,
+                proposal_id,
+                res,
+            ));
+        }
+        Ok(())
+    }
+
+    /// Rejects a multisig proposal
+    pub fn reject_for(
```

Comment:
Out of curiosity, why does it have "for" in the name?

Reply:
No real reason. I added it just to be consistent with `approve_for`. However, there's no real reason for that function to have `for` either :). Since these functions are not extrinsic and hence not directly called by the user, I am guessing that in my mind it was clearer to have an explicit `for` in the name to show that these are indirect calls. To be more in line with our naming convention, I'll change these names to `unsafe_approve` and `unsafe_reject`.